app/javascript/components/widget-controller.jsx

```diff
@@ -56,12 +56,18 @@ export const WidgetController = ({ client, cityName }) => {
       variables={{ id: newWidgetId, cityName }}
       fetchPolicy={fetchPolicy}
     >
-      {({ loading, error, data }) => {
+      {/* eslint-disable-next-line no-shadow */}
+      {({ client, loading, error, data, refetch }) => {
         if (loading) {
           return <LoadingDisplay cityName={cityName} loading={loading} />;
         }
 
         if (error) {
+          setTimeout(async () => {
+            await client.resetStore();
+            await refetch();
```

Comment:
Is it possible to get stuck at these awaits, and the DisconnectedDisplay is never displayed?

Reply:
As it is asynchronous I believe it should show the `<DisconnectedDisplay/>` no matter what as it will not wait for Ln67+68 before returning.
https://developer.mozilla.org/en-US/docs/Web/API/setTimeout#working_with_asynchronous_functions